src/core/Global.ttslua

```diff
@@ -297,6 +297,39 @@ function onPlayerAction(player, action, targets)
       trash.putObject(target)
     end
     return false
+  elseif action == Player.Action.PickUp then
+    local pickedCards = {}
+    for _, target in ipairs(targets) do
+      if target.type == "Card" then
+        table.insert(pickedCards, target)
+      end
+    end
+    if #pickedCards < 6 then
+      for _, pickedCard in ipairs(pickedCards) do
+        local hitList = Physics.cast({
+          origin = pickedCard.getPosition(),
+          direction = {0,1,0},
+          type = 3,
+          size = Vector(2.4, 0, 3.4),
+          orientation = pickedCard.getRotation(),
+          max_distance = 0.5,
+        })
```

Comment:
I'd recommend using our "searchLib" for this. I think this should do the trick:
```lua
local searchResult = searchLib.onObject(pickedCard, "isTileOrToken")
```
That will already include the filtering and return a list of object references.